android/src/main/kotlin/co/doneservices/callkeep/CallKeepNotificationManager.kt

```diff
@@ -0,0 +1,439 @@
+import android.os.Handler
+import android.os.Looper
+import android.text.TextUtils
+import android.view.View
+import android.widget.RemoteViews
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import androidx.core.graphics.drawable.IconCompat
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_ACCENT_COLOR
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_AVATAR
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_DURATION
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_HANDLE
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_ID
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_INCOMING_CALL_NOTIFICATION_CHANNEL_NAME
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_NOTIFICATION_ICON
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_MISSED_CALL_NOTIFICATION_CHANNEL_NAME
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_CALLER_NAME
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_CONTENT_TITLE
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_ACCEPT_TEXT
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_CALLBACK_TEXT
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_DECLINE_TEXT
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_TEXT_MISSED_CALL
+import co.doneservices.callkeep.CallKeepBroadcastReceiver.Companion.EXTRA_CALLKEEP_HAS_VIDEO
+import com.squareup.picasso.OkHttp3Downloader
+import com.squareup.picasso.Picasso
+import com.squareup.picasso.Target
+import okhttp3.OkHttpClient
+
+
+class CallKeepNotificationManager(private val context: Context) {
+
+    companion object {
+
+        const val EXTRA_TIME_START_CALL = "EXTRA_TIME_START_CALL"
+
+        private const val NOTIFICATION_CHANNEL_ID_INCOMING = "callkeep_channel_id"
+        private const val NOTIFICATION_CHANNEL_ID_MISSED = "callkeep_missed_channel_id"
+    }
+
+    private lateinit var notificationBuilder: NotificationCompat.Builder
+    private var notificationViews: RemoteViews? = null
+    private var notificationSmallViews: RemoteViews? = null
+    private var notificationId: Int = 9696
+
+    private var targetLoadAvatarDefault = object : Target {
+        override fun onBitmapLoaded(bitmap: Bitmap?, from: Picasso.LoadedFrom?) {
+            notificationBuilder.setLargeIcon(bitmap)
+            getNotificationManager().notify(notificationId, notificationBuilder.build())
+        }
+
+        override fun onBitmapFailed(e: Exception?, errorDrawable: Drawable?) {
+        }
+
+        override fun onPrepareLoad(placeHolderDrawable: Drawable?) {
+        }
```

Comment:
Do you need to override these even though implementation is empty?

Reply:
I'll check and improve if it's not needed.
Just as a side note, I left Avatar loading logic but we actually don't load it nor view it anywhere currently, we might update our design to use it later so that's why it's left.

Reply:
They are required to be implemented since this is a concrete implementation